source/memory/memory.go

```diff
@@ -30,6 +31,24 @@ type Source struct{}
 // Name returns an identifier string for this feature source.
 func (s Source) Name() string { return "memory" }
 
+var consecutiveSpaces = regexp.MustCompile("[[:space:]]+")
+
+func GetChoice(parameter string) (string, error) {
+	val, err := ioutil.ReadFile(parameter)
+	if err != nil {
+		return "", fmt.Errorf("failed to read '%s': %v", parameter, err)
+	}
+
+	// Split up the choices
+	allChoices := consecutiveSpaces.Split(string(val), -1)
```

Comment:
Shouldn't you just use bytes.Fields() (or strings.Fields()) here?

Reply:
Done.